recipe/meta.yaml

```diff
@@ -1,51 +1,62 @@
 {% set name = "category_encoders" %}
-{% set version = "2.1.0" %}
+{% set version = "2.2.2" %}
 
 package:
   name: {{ name }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 9d76a2d810e02ad9527bed9c70b01fcb0f2238778b5422128275b5ba7561ca64
+  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+    sha256: 79651dc3e8cd6749bc5d404c49a79e93469016d106736326842305e076136cb0
+    folder: dist
+  - url: https://github.com/scikit-learn-contrib/{{ name }}/archive/{{ version }}.tar.gz
+    sha256: b8fa5f2c2f3294ed7847e9c5486566604a6a321308dfb5377440a7ca2a9a5b1f
+    folder: src
 
 build:
   noarch: python
   number: 0
   script:
-    - rm -rf examples  # [unix]
-    - rd /s /q examples  # [win]
-    - "{{ PYTHON }} -m pip install --no-deps --ignore-installed -vvv ."
+    - cd dist && {{ PYTHON }} -m pip install -vvv .
 
 requirements:
   host:
-    - python
+    - numpy >=1.14.0
+    - pandas >=0.21.1
+    - patsy >=0.5.1
     - pip
+    - python >=3.5
+    - scikit-learn >=0.20.0
+    - scipy >=1.0.0
+    - statsmodels >=0.9.0
   run:
-    - numpy >=1.11.3
+    - numpy >=1.14.0
     - pandas >=0.21.1
-    - patsy >=0.4.1
-    - python
+    - patsy >=0.5.1
+    - python >=3.5
     - scikit-learn >=0.20.0
-    - scipy >=0.19.0
-    - statsmodels >=0.6.1
+    - scipy >=1.0.0
+    - statsmodels >=0.9.0
 
 test:
+  source_files:
+    - src/tests
   requires:
-    - nose
-    - python
-    - unittest2
     - cython
+    - pip
+    - pytest
+    - pytest-cov
   imports:
     - category_encoders
   commands:
-    - nosetests category_encoders
+    - python -m pip check
+    - cd src && pytest -vv --cov category_encoders -k "not (pandas_index or truncated_index)"
```

Comment:
Not super excited about the skips, but works locally

Reply:
those two tests were hanging.

<details>
<pre>
=================================== FAILURES ===================================
________________________ TestEncoders.test_pandas_index ________________________

self = <tests.test_encoders.TestEncoders testMethod=test_pandas_index>

    def test_pandas_index(self):
        # see https://github.com/scikit-learn-contrib/categorical-encoding/pull/224
        df = pd.DataFrame({
            'hello': ['a', 'b', 'c'],
            'world': [0, 1, 0]
        }, columns=pd.Index(['hello', 'world']))
        cols = df.select_dtypes(include='object').columns
    
        for encoder_name in (set(encoders.__all__)):
            with self.subTest(encoder_name=encoder_name):
                enc = getattr(encoders, encoder_name)(cols=cols)
>               enc.fit_transform(df, df['world'])

tests/test_encoders.py:665: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/site-packages/sklearn/base.py:574: in fit_transform
    return self.fit(X, y, **fit_params).transform(X)
../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/site-packages/category_encoders/hashing.py:166: in fit
    X_temp = self.transform(X, override_return_df=True)
../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/site-packages/category_encoders/hashing.py:260: in transform
    data = list(hashing_parts.get().values())[0]
<string>:2: in get
    ???
../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/multiprocessing/managers.py:835: in _callmethod
    kind, result = conn.recv()
../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/multiprocessing/connection.py:250: in recv
    buf = self._recv_bytes()
../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/multiprocessing/connection.py:414: in _recv_bytes
    buf = self._recv(4)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <multiprocessing.connection.Connection object at 0x7f843d25a7f0>
size = 4, read = <built-in function read>

    def _recv(self, size, read=_read):
        buf = io.BytesIO()
        handle = self._handle
        remaining = size
        while remaining > 0:
>           chunk = read(handle, remaining)
E           Failed: Timeout >180.0s

../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/multiprocessing/connection.py:379: Failed
----------------------------- Captured stderr call -----------------------------
Process Process-351:
Process Process-349:
Traceback (most recent call last):
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 203, in require_data
    data_part = self.hashing_trick(X_in=data_part, hashing_method=self.hash_method, N=self.n_components, cols=self.cols)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 391, in hashing_trick
    X_cat.columns = new_cols
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 5287, in __setattr__
    return object.__setattr__(self, name, value)
  File "pandas/_libs/properties.pyx", line 67, in pandas._libs.properties.AxisProperty.__set__
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 661, in _set_axis
    self._data.set_axis(axis, labels)
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/internals/managers.py", line 177, in set_axis
    raise ValueError(
ValueError: Length mismatch: Expected axis has 1 elements, new values have 8 elements
Traceback (most recent call last):
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 203, in require_data
    data_part = self.hashing_trick(X_in=data_part, hashing_method=self.hash_method, N=self.n_components, cols=self.cols)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 391, in hashing_trick
    X_cat.columns = new_cols
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 5287, in __setattr__
    return object.__setattr__(self, name, value)
  File "pandas/_libs/properties.pyx", line 67, in pandas._libs.properties.AxisProperty.__set__
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 661, in _set_axis
    self._data.set_axis(axis, labels)
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/internals/managers.py", line 177, in set_axis
    raise ValueError(
ValueError: Length mismatch: Expected axis has 1 elements, new values have 8 elements
Process Process-350:
Traceback (most recent call last):
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 203, in require_data
    data_part = self.hashing_trick(X_in=data_part, hashing_method=self.hash_method, N=self.n_components, cols=self.cols)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 391, in hashing_trick
    X_cat.columns = new_cols
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 5287, in __setattr__
    return object.__setattr__(self, name, value)
  File "pandas/_libs/properties.pyx", line 67, in pandas._libs.properties.AxisProperty.__set__
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 661, in _set_axis
    self._data.set_axis(axis, labels)
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/internals/managers.py", line 177, in set_axis
    raise ValueError(
ValueError: Length mismatch: Expected axis has 1 elements, new values have 8 elements
Process Process-352:
Traceback (most recent call last):
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 203, in require_data
    data_part = self.hashing_trick(X_in=data_part, hashing_method=self.hash_method, N=self.n_components, cols=self.cols)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 391, in hashing_trick
    X_cat.columns = new_cols
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 5287, in __setattr__
    return object.__setattr__(self, name, value)
  File "pandas/_libs/properties.pyx", line 67, in pandas._libs.properties.AxisProperty.__set__
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 661, in _set_axis
    self._data.set_axis(axis, labels)
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/internals/managers.py", line 177, in set_axis
    raise ValueError(
ValueError: Length mismatch: Expected axis has 1 elements, new values have 8 elements
______________________ TestEncoders.test_truncated_index _______________________

self = <tests.test_encoders.TestEncoders testMethod=test_truncated_index>

    def test_truncated_index(self):
        # see: https://github.com/scikit-learn-contrib/categorical-encoding/issues/152
        data = pd.DataFrame(data={'x': ['A', 'B', 'C', 'A', 'B'], 'y': [1, 0, 1, 0, 1]})
        data = data.iloc[2:5]
        data2 = pd.DataFrame(data={'x': ['C', 'A', 'B'], 'y': [1, 0, 1]})
        for encoder_name in encoders.__all__:
            with self.subTest(encoder_name=encoder_name):
                enc = getattr(encoders, encoder_name)()
>               result = enc.fit_transform(data.x, data.y)

tests/test_encoders.py:493: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/site-packages/sklearn/base.py:574: in fit_transform
    return self.fit(X, y, **fit_params).transform(X)
../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/site-packages/category_encoders/hashing.py:166: in fit
    X_temp = self.transform(X, override_return_df=True)
../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/site-packages/category_encoders/hashing.py:260: in transform
    data = list(hashing_parts.get().values())[0]
<string>:2: in get
    ???
../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/multiprocessing/managers.py:835: in _callmethod
    kind, result = conn.recv()
../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/multiprocessing/connection.py:250: in recv
    buf = self._recv_bytes()
../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/multiprocessing/connection.py:414: in _recv_bytes
    buf = self._recv(4)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <multiprocessing.connection.Connection object at 0x7f843d20be50>
size = 4, read = <built-in function read>

    def _recv(self, size, read=_read):
        buf = io.BytesIO()
        handle = self._handle
        remaining = size
        while remaining > 0:
>           chunk = read(handle, remaining)
E           Failed: Timeout >180.0s

../../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/multiprocessing/connection.py:379: Failed
----------------------------- Captured stderr call -----------------------------
Process Process-405:
Process Process-406:
Traceback (most recent call last):
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 203, in require_data
    data_part = self.hashing_trick(X_in=data_part, hashing_method=self.hash_method, N=self.n_components, cols=self.cols)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 391, in hashing_trick
    X_cat.columns = new_cols
Traceback (most recent call last):
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 5287, in __setattr__
    return object.__setattr__(self, name, value)
  File "pandas/_libs/properties.pyx", line 67, in pandas._libs.properties.AxisProperty.__set__
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 661, in _set_axis
    self._data.set_axis(axis, labels)
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/internals/managers.py", line 177, in set_axis
    raise ValueError(
ValueError: Length mismatch: Expected axis has 1 elements, new values have 8 elements
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 203, in require_data
    data_part = self.hashing_trick(X_in=data_part, hashing_method=self.hash_method, N=self.n_components, cols=self.cols)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 391, in hashing_trick
    X_cat.columns = new_cols
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 5287, in __setattr__
    return object.__setattr__(self, name, value)
  File "pandas/_libs/properties.pyx", line 67, in pandas._libs.properties.AxisProperty.__set__
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 661, in _set_axis
    self._data.set_axis(axis, labels)
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/internals/managers.py", line 177, in set_axis
    raise ValueError(
ValueError: Length mismatch: Expected axis has 1 elements, new values have 8 elements
Process Process-408:
Process Process-407:
Traceback (most recent call last):
Traceback (most recent call last):
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "$PREFIX/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 203, in require_data
    data_part = self.hashing_trick(X_in=data_part, hashing_method=self.hash_method, N=self.n_components, cols=self.cols)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 203, in require_data
    data_part = self.hashing_trick(X_in=data_part, hashing_method=self.hash_method, N=self.n_components, cols=self.cols)
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 391, in hashing_trick
    X_cat.columns = new_cols
  File "$PREFIX/lib/python3.8/site-packages/category_encoders/hashing.py", line 391, in hashing_trick
    X_cat.columns = new_cols
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 5287, in __setattr__
    return object.__setattr__(self, name, value)
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 5287, in __setattr__
    return object.__setattr__(self, name, value)
  File "pandas/_libs/properties.pyx", line 67, in pandas._libs.properties.AxisProperty.__set__
  File "pandas/_libs/properties.pyx", line 67, in pandas._libs.properties.AxisProperty.__set__
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 661, in _set_axis
    self._data.set_axis(axis, labels)
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/generic.py", line 661, in _set_axis
    self._data.set_axis(axis, labels)
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/internals/managers.py", line 177, in set_axis
    raise ValueError(
  File "$PREFIX/lib/python3.8/site-packages/pandas/core/internals/managers.py", line 177, in set_axis
    raise ValueError(
ValueError: Length mismatch: Expected axis has 1 elements, new values have 8 elements
ValueError: Length mismatch: Expected axis has 1 elements, new values have 8 elements

</pre>
</details>